src/main/java/com/example/cult_of_tim/cultoftim/service/PromotionService.java

```diff
@@ -15,9 +16,9 @@ public interface PromotionService {
 
     boolean isGlobal(Long promotionID);
 
-    Promotion createPromotion(Promotion promotion);
+    Promotion createPromotion(String description, LocalDateTime startDate, LocalDateTime endDate, List<PromotionDiscount> discounts, boolean globalPromotion);
 
-    Promotion updatePromotion(Long id, Promotion updatedPromotion);
+    Promotion updatePromotion(Long id, String description, LocalDateTime startDate, LocalDateTime endDate, List<PromotionDiscount> discounts, boolean globalPromotion);
```

Comment:
this is just plain wrong, again, https://www.baeldung.com/java-dto-pattern
Or leave it as it was previously, and ask lecturer next time.